### PR TITLE
tests: Add timeout for node sync in queries txsource workload

### PR DESCRIPTION
**Pull request by kostko**

### CODE REVIEW

File: go/oasis-node/cmd/debug/txsource/workload/queries.go [hunk 1]
```diff
@@ -874,13 +876,15 @@ func (q *queries) Run(
...
-			_ = q.control.WaitSync(loopCtx)
-			time.Sleep(1 * time.Second)
-			continue
+		switch isSynced, _ := q.control.IsSynced(loopCtx); isSynced {
+		case true:
+			// Node is synced, proceed with queries.
+			err = q.doQueries(loopCtx, rng)
+		case false:
+			// Node is not synced, wait for it to sync (at most for queriesIterationTimeout).
+			err = q.control.WaitSync(loopCtx)
```
Comment (peternose) [inline go/oasis-node/cmd/debug/txsource/workload/queries.go:885]:
```suggestion
			err = q.control.WaitSync(loopCtx)
			q.iteration--
```

Comment (peternose) [reply go/oasis-node/cmd/debug/txsource/workload/queries.go:885]: If context expires (or any other error occurs), the error will be handled below and the function will return.

**Review by peternose — CHANGES_REQUESTED** [review]

File: go/oasis-node/cmd/debug/txsource/workload/queries.go (hunk 1 above)
Comment (kostko) [reply go/oasis-node/cmd/debug/txsource/workload/queries.go:885]: Yes this is on purpose as the node should not keep syncing forever.

Comment (kostko) [reply go/oasis-node/cmd/debug/txsource/workload/queries.go:885]: I thought about tweaking the iteration count and thought that it shouldn't matter that much, but sure it is better hygiene :)

**Review by peternose — APPROVED** [review]

File: go/oasis-node/cmd/debug/txsource/workload/queries.go (hunk 1 above)
Comment (kostko) [reply go/oasis-node/cmd/debug/txsource/workload/queries.go:886]: I changed it around so the iteration counter is only updated in the normal case.

**Pull request merged**